sdk/rust-examples/aesctr-native/src/main.rs

```diff
@@ -0,0 +1,86 @@
+//! An example program to call the AES Counter mode native module.
+//!
+//! ## Context
+//!
+//! It calls the module mounted at path `/services/aesctr.dat`, via the postcard encoding of
+//! the interface,
+//! ```
+//! pub struct AesCtrInput {
+//!     key: [u8; 16],
+//!     iv: [u8; 16],
+//!     input_path: PathBuf,
+//!     output_path: PathBuf,
+//!     is_encryption: bool,
+//! }
+//! ```
+//!
+//! ## Authors
+//!
+//! The Veracruz Development Team.
+//!
+//! ## Copyright
+//!
+//! See the file `LICENSE_MIT.markdown` in the Veracruz root directory for licensing
+//! and copyright information.
+
+use serde::Serialize;
+use std::{
+    fs::{read, write, File},
+    io::Read,
+    path::{Path, PathBuf},
+};
+
+/// The interface between Counter mode AES service
+#[derive(Serialize, Debug)]
+pub struct AesCtrInput {
+    key: [u8; 16],
+    iv: [u8; 16],
+    input_path: PathBuf,
+    output_path: PathBuf,
+    is_encryption: bool,
+}
+
+/// Example to invoke the Counter mode AES service. Encrypt `data.dat` and then 
+/// decrypt it using the `key.dat` and iv.dat`.
+fn main() -> anyhow::Result<()> {
```

Comment:
Can we tidy this file up a little, e.g., by adding in some new lines between related material, capitalising log messages, and spelling the messages out fully rather than using abbreviations?